.github/workflows/lint.yml

```diff
@@ -0,0 +1,38 @@
+---
+name: Project Validation
+on:
+  push:
+    branches-ignore: [ main ]
+jobs:
+  lint:
```

Comment:
Rename to `validate` as this job does more than just linting.

Also consider replacing the workflow with a reusable workflow and a engineering image so that we can maintain tooling in fewer locations.

Using an engineering image also allows us to run all the CI tools locally.

Reply:
Room is open for contributes, I cannot cover all stuff by my own in one PR